### ISSUE TITLE
[ADOC] DAPS can not handle xrefs to other documents

### ISSUE THREAD

**Issue by r0ckarong** [openSUSE/daps#508]
### Problem description

In Geekodoc you can define an xref to another document because the way DAPS and the config files work these documents are considered a "Set". DAPS can "know" the content of other documents in the set and generate links or spelled out section titles for use as a cross reference.

When using AsciiDoc as input format the "Set" concept does not apply and DAPS will complain about a missing `IDREF` because the section ID referred to is not in the same document.

### Expected behavior

DAPS will do some processing of the docs in the content directory and find this reference and will generate a suitable link (or spelled out instruction how to find the section for PDF).

Duplicate IDs will be very possible but can be worked against by defining manual section IDs in the AsciiDoc files.

### Steps to reproduce the behavior

Set an xref to a section not in the document.
Build file

```
/home/nora.koranova/git/doc-caasp/build/.profiled/noprofile/book_admin.xml:83:49: error: IDREF "_book.caasp.installquick" without matching ID
/home/nora.koranova/git/doc-caasp/build/.profiled/noprofile/book_admin.xml:90:42: error: IDREF "_book.caasp.admin" without matching ID
/home/nora.koranova/git/doc-caasp/build/.profiled/noprofile/book_admin.xml:75:47: error: IDREF "_book.caasp.deployment" without matching ID
```

**Comment by ghost** [comment]
I think I lack the creativity to see a way for this to actually work. :)
For better or worse, I think Antora resolves this issue somehow -- do you know how?

> Duplicate IDs will be very possible but can be worked against by defining manual section IDs in the AsciiDoc files.

Even if IDs are set manually that does not mean there is not a case where you'd have duplicate IDs:
E.g. in doc-sle we are actually intentionally using duplicate IDs with different profiling  (~ ifdefs in AsciiDoc) in some cases where we need similar sections for both SLES and SLED. In that case, only profiling makes the document valid then. But if documents use different DC files applying the same profiling across them is weird at best. (I hope that makes sense.)

**Comment by r0ckarong** [comment]
> I think Antora resolves this issue somehow -- do you know how?

I would think they parse the entire contents of the modules and establish some context so if a reference shows up they can produce a proper link. The "content classifier" sounds like it would allow the usage of such information. https://gitlab.com/antora/antora/tree/master/packages/content-classifier

**Comment by fsundermeyer** [comment]
The problem is not DAPS, but rather ASCIIDoc, since it does not support sets but only stand-alone documents.

**Comment by fsundermeyer** [comment]
Fixed with 4f145320dcbc
